src/extensions/endpoints/adoption-code/src/repositories/directus.ts

```diff
@@ -59,11 +60,28 @@ export const createAdoptedProbe = async (userId: string, probe: ProbeToAdopt, co
 			sendNotificationProbeAdopted({ ...adoption, id }, context),
 			existingProbe.userId && existingProbe.userId !== userId && sendNotificationProbeUnassigned(existingProbe, context),
 		]);
-	} else {
-		id = await itemsService.createOne(adoption, { emitEvents: false });
-		await sendNotificationProbeAdopted({ ...adoption, id }, context);
+
+		return [ id, name ] as const;
+	}
+
+	// Probe not found by ip/uuid, trying to find user's offline probe by city/asn.
+	const probeByAsn = await database('gp_probes')
+		.where({
+			userId: adoption.userId,
+			status: 'offline',
+			asn: adoption.asn,
+			city: adoption.city,
+		})
```

Comment:
Let's additionally order these by the most recent activity to make the behavior deterministic.